web/auth.go

```diff
@@ -14,33 +15,31 @@ func (s *Server) Register(c *gin.Context) {
 		Timestamp string `json:"timestamp"`
 		Signature string `json:"signature"`
 		Requester string `json:"requester"`
+		EncKey    string `json:"encryption_public_key"`
 	}
 
 	if err := c.BindJSON(&req); err != nil {
-		abortWithEncoding(c, http.StatusBadRequest, ErrorResponse{
-			Message: err.Error(),
-		}, err)
+		abortWithErrorMessage(c, http.StatusBadRequest, errorResponse{Message: err.Error()}, err)
+		return
+	}
+	sig, err := hex.DecodeString(req.Signature)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"reason": "signature not hex-encoded"})
+		return
+	}
+	if err := account.Verify(req.Requester, []byte(req.Timestamp), sig); err != nil {
```

Comment:
`EncKey` should be a part of the signature as well. Otherwise, there will be a replay attack happened.